README.md

```diff
@@ -146,6 +146,41 @@ Default values can be overridden on a request-by-request basis.
 When a default `url:` key is specified, a request's full URL will be built by
 joining the base `default url: ...` value with the request's `path:` option.
 
+In this example, `ArticlesClient.configuration` will read directly from the
+environment-aware `config/clients/articles.yml` file.
+
+Consider the following configuration:
+
+```yaml
+# config/clients/articles.yml
+default: &default
+  url: "https://staging.example.com"
+
+development:
+  <<: *default
+
+test:
+  <<: *default
+
+production:
+  <<: *default
+  url: "https://example.com"
+```
+
+Then from the client class, read those values directly from `configuration`:
+
+```ruby
+class ArticlesClient < ActionClient::Base
+  default url: configuration.url
```

Comment:
Would it be too magical to make this the default behaviour?

We'd look for:

1. An explicit `default url`,
2. fall back to `configuration.url`,
3. fall back to `nil` (which IIRC expects full URLs to be passed in each request method).

Reply:
I think those would be logical next steps in this feature's future. @composerinteralia and I went back and forth on what we'd want to do on a consumer's behalf, and felt that might be too magical.

We were also considering some [generators](https://github.com/thoughtbot/action_client/issues/6) that could help edify those conventions, so maybe it'll feel less surprising in the future.